codegen/winterfell/src/air/graph.rs

```diff
@@ -68,15 +68,23 @@ impl Codegen for Operation {
                     matrix_access.col_idx()
                 ),
             },
-            Operation::TraceElement(trace_access) => match trace_access.row_offset() {
-                0 => {
-                    format!("current[{}]", trace_access.col_idx())
-                }
-                1 => {
-                    format!("next[{}]", trace_access.col_idx())
-                }
-                _ => panic!("Winterfell doesn't support row offsets greater than 1."),
-            },
+            Operation::TraceElement(trace_access) => {
+                let frame = if let 0 = trace_access.trace_segment() {
+                    "main"
+                } else {
+                    "aux"
+                };
+                let row_offset = match trace_access.row_offset() {
+                    0 => {
+                        format!("current[{}]", trace_access.col_idx())
+                    }
+                    1 => {
+                        format!("next[{}]", trace_access.col_idx())
+                    }
+                    _ => panic!("Winterfell doesn't support row offsets greater than 1."),
+                };
+                format!("{}_{}", frame, row_offset)
```

Comment:
This works - but I think implementing `Codegen` on `IndexedTraceAccess` would be cleaner.

Reply:
Makes sense. Thanks. Done.